tests/kernel/mem_protect/mem_protect/src/main.c

```diff
@@ -90,7 +107,12 @@ void test_main(void)
 			 ztest_unit_test(test_create_new_supervisor_thread_from_user),
 			 ztest_unit_test(test_create_new_essential_thread_from_user),
 			 ztest_unit_test(test_create_new_higher_prio_thread_from_user),
-			 ztest_unit_test(test_create_new_invalid_prio_thread_from_user)
+			 ztest_unit_test(
+				 test_create_new_invalid_prio_thread_from_user),
+			 /* Due to the generated assertion error, test
+			  * test_mem_part_assert_add_overmax must be in the end
+			  */
+			 ztest_unit_test(test_mem_part_assert_add_overmax)
```

Comment:
don't test against asserts. file a bug instead.

Reply:
Seems that there is no bug, I put that test before the last one and assertion works correctly. That test verifies that assertion in zephyr/kernel/mem_domain.c L171 works as expected.

Reply:
https://github.com/zephyrproject-rtos/zephyr/issues/26621 I submitted an issue.